store/tikv/mock-tikv/mvcc_test.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mocktikv
+
+import (
+	"testing"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/tidb/util/codec"
+)
+
+func TestT(t *testing.T) {
+	TestingT(t)
+}
+
+type testMockTiKVSuite struct {
+	store *MvccStore
+}
+
+var _ = Suite(&testMockTiKVSuite{})
+
+func encodeKey(s string) []byte {
+	return codec.EncodeBytes(nil, []byte(s))
+}
+
+func encodeKeys(ss []string) [][]byte {
+	var keys [][]byte
+	for _, s := range ss {
+		keys = append(keys, encodeKey(s))
+	}
+	return keys
+}
+
+func putMutations(kvpairs ...string) []*kvrpcpb.Mutation {
+	var mutations []*kvrpcpb.Mutation
+	for i := 0; i < len(kvpairs); i += 2 {
+		mutations = append(mutations, &kvrpcpb.Mutation{
+			Op:    kvrpcpb.Op_Put.Enum(),
+			Key:   encodeKey(kvpairs[i]),
+			Value: []byte(kvpairs[i+1]),
+		})
+	}
+	return mutations
+}
+
+func (s *testMockTiKVSuite) SetUpTest(c *C) {
+	s.store = NewMvccStore()
+}
+
+func (s *testMockTiKVSuite) mustGetNone(c *C, key string, ts uint64) {
+	val, err := s.store.Get(encodeKey(key), ts)
+	c.Assert(err, IsNil)
+	c.Assert(val, IsNil)
+}
+
+func (s *testMockTiKVSuite) mustGetErr(c *C, key string, ts uint64) {
+	val, err := s.store.Get(encodeKey(key), ts)
+	c.Assert(err, NotNil)
+	c.Assert(val, IsNil)
+}
+
+func (s *testMockTiKVSuite) mustGetOK(c *C, key string, ts uint64, expect string) {
```

Comment:
@shenli This tests and functions are ported from TiKV. I think it would be helpful to maintain the same structure.